src/marqo/core/inference/inference_models/hugging_face_model_properties.py

```diff
@@ -0,0 +1,119 @@
+            We are not raising an error right now as that would be a breaking change.
+        tokens: The token length of the model. It is default to 128.
+        type: The type of the model. It should be "hf".
+        url: The URL of the model checkpoint. It is optional.
+        dimensions: The dimensions of the model.
+        model_location: The location of the model. It is optional.
+        note: A note about the model. It is optional.
+        pooling_method: The pooling method for the model. It should be one of the values in the PoolingMethod enum.
+    """
+    name: Optional[str] = None
+    tokens: int = 128
+    type: str
+    dimensions: int = Field(..., ge=1)
+    url: Optional[str] = None
+    model_location: Optional[ModelLocation] = Field(default=None, alias="modelLocation")
+    note: Optional[str] = None
+    pooling_method: PoolingMethod = Field(..., alias="poolingMethod")
+
+    @validator("type")
+    def _validate_type(cls, v):
+        if v != "hf":
+            raise ValueError("The type of the model should be 'hf'.")
+        return v
+
+    @root_validator(pre=True, skip_on_failure=True)
+    def _validate_or_infer_pooling_method(cls, values):
+        """Infer the pooling_method from the model name if it is not provided.
+
+        If the pooling_method is provided, return the values as is.
+        """
+        pooling_method = values.get("pooling_method") or values.get("poolingMethod")
+        if pooling_method is not None:
+            return values
+        name = values.get('name')
+        if isinstance(name, str) and name:
+            pooling_method =  cls._infer_pooling_method_from_name(name)
+        else:
+            pooling_method = PoolingMethod.Mean
+        values["pooling_method"] = pooling_method
+        return values
+
+    @staticmethod
+    def _infer_pooling_method_from_name(name: str) -> PoolingMethod:
+        """
+        Infer the pooling method from the model name.
+        Args:
+            name: The name of the model. This is the repo_id in the Hugging Face model hub.
+
+        Returns:
+            The inferred pooling method.
+        """
+        repo_id = name
+        file_name = "1_Pooling/config.json"
+        try:
+            file_path = hf_hub_download(repo_id, file_name, cache_dir=ModelCache.hf_cache_path)
+        except HfHubHTTPError:
+            logger.warn(f"Could not infer pooling method from the model {name}. Defaulting to mean pooling.")
+            return PoolingMethod.Mean
+
+        try:
+            with open(file_path, 'r') as file:
+                content = json.loads(file.read())
+        except JSONDecodeError:
+            logger.warn(f"Could not infer pooling method from the model {name}. Defaulting to mean pooling.")
+            return PoolingMethod.Mean
+
+        if not isinstance(content, dict):
+            logger.warn(f"Could not infer pooling method from the model {name}. Defaulting to mean pooling.")
```

Comment:
The code snippet:
```
logger.warn(f"Could not infer pooling method from the model {name}. Defaulting to mean pooling.")
            return PoolingMethod.Mean
```

Is repeated a lot. It could be split into a function, or put at the bottom of this function and triggered with a boolean flag